### PR TITLE
Add support for burning junk and shields

### PR DESCRIPTION
**Pull request by brian-man**
This primary focus of this change is:
- add ISalvageable interface
- add IReforgeable interface
- add ISalvageable support to Junk
- add ISalvageable support to Shields
- add IReforgeable support to Weapons
- add singular and bulk burnJunk to game
- add burnShield and burnShields to game

Additional improvements:
- add isOwner(nft,id) and isOwnerOfMultiple(nft,ids) modifiers
- remove isCharacterOwner, isWeaponOwner, and isWeaponsOwner
- clean up unused parameter warning in multiple _beforeTokenTransfer implementations

### CODE REVIEW

**Review by kcper — COMMENTED** [review]
Looks ok to me.

**Comment by kcper** [comment]
Also there are some rather simple to resolve conflicts.

File: contracts/cryptoblades.sol [hunk 1]
```diff
@@ -612,13 +647,23 @@ contract CryptoBlades is Initializable, AccessControlUpgradeable {
...
-    function _burnWeaponsLogic(uint256[] memory burnIDs) internal {
+    function _burnReforgeablesLogic(IReforgeable reforgeable, uint256[] memory burnIDs) internal {
         for(uint i = 0; i < burnIDs.length; i++) {
-            weapons.burn(burnIDs[i]);
+            weapons.burnReforgeable(reforgeable, burnIDs[i]);
+        }
+    }
+
+    function _burnSalvageableLogic(ISalvageable salvageable, uint256 burnID) internal {
+        weapons.burnSalvageable(salvageable, burnID);
```
[the remark this answers is not in the document]
Comment (poshdan) [reply contracts/cryptoblades.sol:656]: 1. yeah the weapons contract should not be bloated by other nft burns. There's a function to increment the dust supply externally
2. the nft contracts should not be initialized as variables in the main contract. Shields are already in blacksmith as a start. I'd rather have new functions there
3. the interface seems overkill for now, especially IReforgable with its whopping one use

**Comment by kcper** [comment]
I think this can be closed for now. With the introduction of Burning Manager this is a bit outdated, but still some patterns may be reused in the scope of burning manager.

**Pull request closed without merging**